schema/snapcraft.yaml

```diff
@@ -202,6 +202,27 @@ properties:
             description: What kind of wrapper to generate for the given command
             enum:
               - none
+          sockets:
+            type: object
+            additionalProperties: false
+            patternProperties:
+              "^[a-z0-9][a-z0-9+-]*$":
```

Comment:
This pattern is supposed to correspond to [this](https://github.com/snapcore/snapd/blob/61ee25526802e6866c7645f53812b43eedc521dc/snap/validate.go#L32), no? The pattern I see here would allow _just_ numbers, for example.

Reply:
Yes, it should correspondo to that. I copied this from the snap `name`, attribute; doesn't that mean that a snap name with just numbers would also be allowed?

Reply:
Yes indeed, it appears snapcraft and snapd have diverged on this point. I suggest using the correct pattern here, I'll fix the snap name in another PR.